File: src/main/java/me/oddlyoko/odoo/modules/trackers/OdooModuleModificationTracker.java
package me.oddlyoko.odoo.modules.trackers;

import com.intellij.openapi.util.SimpleModificationTracker;

import java.util.HashMap;
import java.util.Map;

public class OdooModuleModificationTracker extends SimpleModificationTracker {
    private static final Map<String, OdooModuleModificationTracker> trackers = new HashMap<>();

    public static OdooModuleModificationTracker get(String module) {
        OdooModuleModificationTracker tracker = trackers.get(module);
        if (tracker == null) {
            tracker = new OdooModuleModificationTracker();
            trackers.put(module, tracker);
        }
        return tracker;
    }
}

File: src/main/kotlin/me/oddlyoko/odoo/modules/trackers/OdooModuleModificationTracker.kt
package me.oddlyoko.odoo.modules.trackers

import com.intellij.openapi.util.SimpleModificationTracker

object OdooModuleModificationTracker {
    private val trackers: MutableMap<String, SimpleModificationTracker> = HashMap()

    operator fun get(module: String) = trackers.getOrElse(module) {
        val tracker = SimpleModificationTracker()
        trackers[module] = tracker
        return tracker
    }
}
